relayer/src/chain/handle.rs

```diff
@@ -34,13 +34,16 @@ use ibc_proto::ibc::core::client::v1::QueryConsensusStatesRequest;
 use ibc_proto::ibc::core::commitment::v1::MerkleProof;
 pub use prod::ProdChainHandle;
 
-use crate::connection::ConnectionMsgType;
-use crate::keyring::KeyEntry;
-use crate::{error::Error, event::monitor::EventBatch};
+use crate::{
+    connection::ConnectionMsgType,
+    error::Error,
+    event::monitor::{Error as EventMonitorError, EventBatch},
+    keyring::KeyEntry,
+};
 
 mod prod;
 
-pub type Subscription = channel::Receiver<Arc<EventBatch>>;
+pub type Subscription = channel::Receiver<Arc<Result<EventBatch, EventMonitorError>>>;
```

Comment:
Would it be good to make `Arc<Result<EventBatch, EventMonitorError>>` a type of its own? Even better might be to have `Result<EventBatch, EventMonitorError>` a type of its own, since this has more potential for reuse. Something like `put type MonitorResult = Result<EventBatch, EventMonitorError>`. (Feel free to ignore the comment if it is more idiomatic to retain the `Result` as it is.)

I found the modifications to the monitor very difficult to understand, mostly because of the complex interactions between Monitor, Runtime, and Handler.
There are already quite a few types at the interface with the event monitor. I noticed the following:

- in file handle.rs:
    - `pub type Subscription = channel::Receiver<Arc<Result<EventBatch, EventMonitorError>>>;`
- in file monitor.rs:
    - `type SubscriptionResult = RpcResult<RpcEvent>;`
    - `type SubscriptionStream = dyn Stream<Item = SubscriptionResult> + Send + Sync + Unpin;`
    - `pub type Result<T> = std::result::Result<T, Error>;`
    - `pub type EventReceiver = channel::Receiver<Result<EventBatch>>;`


We can ignore `SubscriptionResult` and `SubscriptionStream` since these seem internal to the monitor.
If I understand correctly, an oversimplification of the flow of events flow is like this:

```

| Monitor |
     |---- Result<EventBatch, EventMonitorError> ---> | Runtime | 
                                                           |
                                                           |
                                   Arc<Result<EventBatch, EventMonitorError> 
                                                           |
                                                           |----> | Handler |

```

An ADR would be nice.

Reply:
This should have read:

```rust
pub type Subscription = channel::Receiver<Arc<Result<EventBatch>>;
```

which is a bit more digestible imho. But yes we could go further and make an alias for `Result<EventBatch>`.

At the same time, all these aliases make me think that there maybe something wrong with the design, and I would like to refactor the whole thing anyway.

> Something like put type MonitorResult = Result<EventBatch, EventMonitorError>. 

I think internally it's customary to stick with `Result` and only rename it when imported in another module, but we could still do that.

> flow of events

At the moment there is still an event bus between the runtime and the supervisor, but we could maybe get rid of it (and the `Arc` wrapper) once we only have a single supervisor. We might need those events in other places so it's not clear if we can actually remove the event bus. To be investigated.

Reply:
Thanks for thorough answers. ⬆️ 
Agree with everything, then let's keep the types as they are (except for the low-hanging fruit of `Subscription`), and seems like the monitor is a good target for refactoring sometime soon. Would like to get a chunk of that!